packages/ui/src/proposals/components/ProposalStages.tsx

```diff
@@ -15,31 +16,48 @@ export interface ProposalStagesProps extends ControlProps<number> {
 export const ProposalStages = ({ status, updates, constitutionality = '-', value, onChange }: ProposalStagesProps) => {
   const rounds = useMemo(() => {
     const decidingCount = updates.filter(({ status }) => status === 'deciding').length
+
+    const totalRoundCount = typeof constitutionality === 'string' ? undefined : constitutionality
+
     const lastUpdate = last(updates).status
     const onGoing = lastUpdate === status
     const approved = lastUpdate === 'gracing'
     const rejected = !onGoing && !approved
     const isDeciding = onGoing && status === 'deciding'
     const isDormant = onGoing && status === 'dormant'
+
     return [
       ...repeat((round) => ({ icon: <CheckboxIcon />, onClick: () => onChange(round) }), decidingCount - 1),
       {
         icon: !isDeciding && (rejected ? <CrossIcon /> : <CheckboxIcon />),
         onClick: () => onChange(decidingCount - 1),
       },
-      ...(isDormant ? [{ icon: false, onClick: undefined }] : []),
+      ...(isDormant
+        ? repeat(
+            () => ({ icon: false, onClick: undefined }),
+            totalRoundCount !== undefined ? totalRoundCount - decidingCount - 1 : 1
+          )
+        : []),
```

Comment:
typeof constitutionality === "string" means if constitutionality === "-" then undefined else if constitutionality === number then constituonality.

Reply:
I see. That was a bad solution on my end :sweat: . 
It should be:
```ts
export interface ProposalStagesProps extends ControlProps<number> {
  rounds: VotingRound[]
  status: ProposalStatus
  constitutionality?: number
}

export const ProposalStages = ({ rounds, status, constitutionality, value, onChange }: ProposalStagesProps) => {
```

and `Council approvals {round + 1}/{constitutionality ?? '-'}`

This way the return becomes:
```ts
repeat(
  () => ...,
  constitutionality ?? 1
)
```
no need for `totalRoundCount`

Reply:
![image](https://user-images.githubusercontent.com/101491745/217604275-752ad643-8b7c-4eca-9785-3fac1ce245b6.png)
There are errors when I try to change the code as you suggested.

Reply:
Yes it looks like I wrote `rounds` instead of `updates`. But still the current code should be changed since it doesn't achieve what we want on the deciding stage. Because now we want to always show all the tabs (although some of them might be disabled). The number of tabs will always be `constitutionality`. So there should be only one `repeat`.

About my second comment I was proposing to change the type of the items returned by this repeat from: `{ icon: boolean, onClick: undefined | () => void }` **to**: `'approved' | 'rejected' | 'deciding' | 'disabled'` in order to make the code more readable.